scrapy_zyte_api/_middlewares.py

```diff
@@ -108,7 +108,9 @@ def process_request(self, request, spider):
         slot.delay = 0
 
         if self._max_requests_reached(downloader):
-            self._crawler.engine.close_spider(spider, "closespider_max_zapi_requests")
+            self._crawler.engine.close_spider(
+                spider, f"closespider_max_{self._max_requests}_zapi_requests"
```

Comment:
I think it's better to have the close reasons static, so that they are not unique each run. If we need any additional information, then stats look like a better place.